src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py

```diff
@@ -543,13 +553,35 @@ def preserve_envvars(spawner):
     return profile
 
 
+def parse_roles(data):
+    parsed_roles = {}
+
+    for role in data:
+        for group in role["groups"]:
+            # group = str(group).replace("/", "")
+            group_name = Path(group).name
+            if group_name not in parsed_roles:
+                parsed_roles[group_name] = []
+
+            role_info = {
+                "description": role["description"],
+                "name": role["name"],
+                "attributes": role["attributes"],
+            }
+
+            parsed_roles[group_name].append(role_info)
+
+    return parsed_roles
+
+
 @gen.coroutine
 def render_profiles(spawner):
     # jupyterhub does not yet manage groups but it will soon
     # so for now we rely on auth_state from the keycloak
     # userinfo request to have the groups in the key
     # "auth_state.oauth_user.groups"
     auth_state = yield spawner.user.get_auth_state()
+    group_roles = parse_roles(spawner.authenticator.group_roles_map)
```

Comment:
This is going to cause problem in jhub-apps: https://github.com/nebari-dev/jhub-apps/blob/657f7c871003e5f678d865b2b4aad1845e3b5bf7/jhub_apps/service/utils.py#L53

Since that's a downstream problem, it's not blocking this but I would need to think of an alternative approach in jhub-apps to get the spawner with `authenticator`.

Reply:
I can add the extra information to the auth_state object (probably), this way you don't need to customize the above mock spawner, and we will not need to rely on `spawner.authenticator` here (which I think its messy). 

Though, the reason I didn't go for it was that I was skeptical of changing it directly as 
- changes might not fully apply
- It looked like that object was moved across different location in the jupyter code during the spawn process

Reply:
I will have a try on that, and check how it goes

Reply:
I'll check if there is a less hacky way to do this on the jhub-apps.

Reply:
Actually there isn't a non-hacky way to do this in jhub-apps. In native JupyterHub the spawner object is magically available due to the huge initialisation done at the startup.

Ref:
- https://github.com/jupyterhub/jupyterhub/blob/f4fa229645e007f3c3ea9d2ed440855f7b19595f/jupyterhub/app.py#L3189 (this initialisation, makes the later possible)
- https://github.com/jupyterhub/jupyterhub/blob/f4fa229645e007f3c3ea9d2ed440855f7b19595f/jupyterhub/handlers/pages.py#L155

We do need to set the groups via `auth_state` and in-fact I think that's even cleaner as all the roles and scopes parsing happens in the authenticator and other components like spawner just consumes the output via `auth_state`.